### PR TITLE
Update xunit assert 2.5.0

### PR DESCRIPTION
**Pull request by agocke**
Xunit 2.5 is out, so bring the XUnit.assert fork up to date. This also brings the tests over from the xunit/xunit repo.

### CODE REVIEW

**Comment by agocke** [comment]
Note: the subtree isn't working properly because the commit is being squashed into the root. I may have to use admin permissions to force merge this via merge commit.

**Comment by ViktorHofer** [comment]
What are the manual changes that you want me to review? Asking as the diff is quite large.

**Comment by agocke** [comment]
@ViktorHofer good point. The only changes I made are https://github.com/dotnet/arcade/pull/13927/commits/5e8d78edf408ac62d5e25714e136c339e394c83f and the previous commit, and the only things I did were add annotations

**Review by ViktorHofer — APPROVED** [review]

File: src/Microsoft.DotNet.XUnitAssert/tests/Microsoft.DotNet.XUnitAssert.Tests.csproj [hunk 1]
```diff
@@ -12,4 +12,8 @@
     <ProjectReference Include="..\src\Microsoft.DotNet.XUnitAssert.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Include="nsubstitute" Version="4.3.0" />
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.XUnitAssert/tests/Microsoft.DotNet.XUnitAssert.Tests.csproj:16]: Please move this version to Versions.props.

**Comment by agocke** [comment]
@mmitche This is just about ready to go, but `git subtree` does not like squash commits. Could you take a look and force merge it through using a proper merge commit? Or alternatively lift the requirement that everything must be squashed?

**Comment by agocke** [comment]
Test failures looks unrelated

**Comment by ViktorHofer** [comment]
@agocke thanks for working on this. Is the code synced up to 2.5.0 or 2.5.1? We already use 2.5.1.

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**